src/routing/http-routing-v1.md

```diff
@@ -106,6 +106,60 @@ Each object in the `Providers` list is a record conforming to a schema, usually
 
 ## Peer Routing API
 
+### `GET /routing/v1/closest-peers/{peer-id}?[closerThan]&[count]`
```

Comment:
@achingbrain I think we could prototype this in https://github.com/ipfs-shipyard/someguy and deploy to [https://delegated-ipfs.dev](https://docs.ipfs.tech/concepts/public-utilities/#delegated-routing). 

Priority-wise, and for the purpose of IPIP we will write for this at some point, does the utility here stops at finding Circuit Relay servers viable in browsers without being a DHT client? 



Would it be useful to have Amino-specific API where the entire hash space could be queried for closest value, not just PeerIDs?
I wonder if this be generalized into delegated `/routing/v1/closest/{cid}?[than]&[count]`, or be limited to peers.

Reply:
> does the utility here stops at finding Circuit Relay servers viable in browsers without being a DHT client?

No, there are other uses - you may wish to use this to populate your routing tables, or to find peers that can host provider records.  It would be useful to perform these operations from environments that have limited resources or cannot open many concurrent connections, for example.

> Would it be useful to have API where the entire hash space could be queried for closest value, not just PeerIDs?

Yes, plus here the PeerIDs are represented as CIDs, so the above is something that can be performed.

To make this clearer it might be better specced as `{cid}` rather than `{peer-id}` I guess.

Reply:
> does the utility here stops at finding Circuit Relay servers

Note: In many cases you'd want relay servers you'd also want peer routing (i.e. finding your address based on your peerID) which in the case of Amino means having recently pinged the closest peers to your peerID in XOR space. Ideally one of those (20) nodes is also acting as your relay which means the number of connections and queries you'd need are even smaller.